zkchannels-contract/zkchannel_smartpy_script.py

```diff
@@ -40,89 +39,113 @@ def is_g1_not_zero(val):
-                  delayExpiry       = sp.timestamp(0),
-                  g2                = g2,
-                  merchPk0          = merchPk0,
-                  merchPk1          = merchPk1,
-                  merchPk2          = merchPk2,
-                  merchPk3          = merchPk3,
-                  merchPk4          = merchPk4,
-                  merchPk5          = merchPk5,
+                  cid               = cid,              # the unique identifier for the channel.
+                  custAddr          = custAddr,         # customer tz1 address
+                  merchAddr         = merchAddr,        # merchant tz1 address
+                  custPk            = custPk,           # customer tezos public key
+                  merchPk           = merchPk,          # merchant tezos public key
+                  custBal           = sp.mutez(0),      # customer's closing balance
+                  merchBal          = sp.mutez(0),      # merchant's closing balance
+                  custFunding       = custFunding,      # customer's initial balance
+                  merchFunding      = merchFunding,     # merchant's initial balance
+                  status            = sp.nat(AWAITING_CUST_FUNDING), # contract status
+                  revLock           = sp.bytes("0x00"), # revLock initialized to 0x00
+                  selfDelay         = selfDelay,        # delay in seconds
+                  delayExpiry       = sp.timestamp(0),  # if the delay is triggered, delayExpiry records when the delay is due to expire
+                  g2                = g2,               # PS pubkey
+                  merchPk0          = merchPk0,         # PS pubkey
+                  merchPk1          = merchPk1,         # PS pubkey
+                  merchPk2          = merchPk2,         # PS pubkey
+                  merchPk3          = merchPk3,         # PS pubkey
+                  merchPk4          = merchPk4,         # PS pubkey
+                  merchPk5          = merchPk5,         # PS pubkey
                   close_flag        = close_flag,
                   context_string    = sp.string("zkChannels mutual close"))
- 
-    # addFunding is called by the customer or the merchant to fund their
-    # portion of the channel (according to the amounts specified in custFunding
-    # and merchFunding). The full amount must be funded in one transaction.
+
+    # addFunding is called by the customer or the merchant to fund their portion of 
+    # the channel (according to the amounts specified in custFunding and merchFunding). 
+    # The full amount must be funded in one transaction. The customer must fund their 
+    # side of the channel before the merchant.
     @sp.entry_point
     def addFunding(self):
-        sp.verify(self.data.status == AWAITING_FUNDING)
+        # Only allow the customer or merchant to fund the contract.
         sp.verify((self.data.custAddr == sp.sender) | (self.data.merchAddr == sp.sender))
```

Comment:
would it make sense to just have different entrypoints for adding funding: one for the customer and one for the merchant?

Reply:
We could create separate AddFunding entrypoints for the customer and merchant. The only issue is it would be a breaking change for the demo and a lot of the tests. I could update them if we make the change but I just wanted to check if you think it's worth it?

Reply:
note: we decided it in slack it's worth it. 
Now we have addCustFunding and addMerchFunding